src/components/ScreenWrapper/index.js

```diff
@@ -37,9 +36,14 @@ class ScreenWrapper extends React.Component {
             Navigation.dismissModal();
         }, shortcutConfig.descriptionKey, shortcutConfig.modifiers, true);
 
-        this.unsubscribeTransitionEnd = onScreenTransitionEnd(this.props.navigation, () => {
+        this.unsubscribeTransitionEnd = this.props.navigation.addListener('transitionEnd', () => {
```

Comment:
I think we should be using [onScreenTransitionEnd](https://github.com/Expensify/App/tree/main/src/libs/onScreenTransitionEnd) and add the `onScreenTransitionStart` method. There seems to be an [issue](https://github.com/Expensify/App/blob/e43bab50d8b61efec1d7410f4019806abbd1cdcd/src/libs/onScreenTransitionEnd/index.ios.js#L2) on iOS with using `navigation.addListener('transitionEnd'`

Reply:
Upon reviewing the issue https://github.com/software-mansion/react-native-screens/issues/713, it appears that the issue of the "transitionEnd" event not firing upon calling the "navigation.goBack()" method has been addressed. Our team had previously implemented a workaround utilizing InteractionManager, however, it appears that this issue has been resolved in more recent versions of React Native and React Native Reanimated. Through testing, I have determined that the "transitionEnd" and "transitionStart" events are now functioning as expected when utilizing the "navigation.goBack()" method. This issue was encountered when our team was utilizing  `react-native: 0.64.1` and `react-native-reanimated : ^2.3.0-alpha.1` and it appears to have been resolved in more recent versions. 

Could you please test it verify if you find an edge case where navigation.addListener('transitionEnd') and navigation.addListener('transitionStart') doesn't fire up.

Reply:
Ok, I've found the [PR](https://github.com/Expensify/App/pull/2956) that introduced this workaround for iOS, tested without it and the animations are still smooth, so I think you're right, the issue seems to have been resolved.
In that case, we can just delete [onScreenTransitionEnd/index.ios.js] (https://github.com/Expensify/App/blob/main/src/libs/onScreenTransitionEnd/index.ios.js) (and make it a single file instead of a folder)
And implement the onScreenTransitionStart method alongside it

Reply:
Ok, Sounds good I will make changes in next minutes.